yt_dlp/extractor/owncloud.py

```diff
@@ -0,0 +1,88 @@
+import urllib.parse
+
+from yt_dlp.extractor.common import InfoExtractor
+from yt_dlp.utils import (
+    determine_ext,
+    ExtractorError,
+    url_or_none,
+    urlencode_postdata,
+)
+
+
+class OwnCloudIE(InfoExtractor):
+    _INSTANCES_RE = r'''(?:
+                            sciebo\.de|
+                            [^\.]+\.sciebo\.de|
+                            cloud\.uni-koblenz-landau\.de|
+                        )'''
+    _VALID_URL = rf'''(?x)
+        (?P<server>https?://{_INSTANCES_RE})/s/
+        (?P<id>[\w\-\.]+)'''
+
+    _TESTS = [
+        {
+            'url': 'https://ruhr-uni-bochum.sciebo.de/s/wWhqZzh9jTumVFN',
+            'info_dict': {
+                'id': 'wWhqZzh9jTumVFN',
+                'ext': 'mp4',
+                'title': 'CmvpJST.mp4',
+            },
+        },
+    ]
+
+    def _real_extract(self, url):
+        server, video_id = self._match_valid_url(url).group('server', 'id')
+
+        webpage, urlh = self._download_webpage_handle(f'{server}/s/{video_id}', video_id, 'Downloading webpage')
+
+        if self._search_regex(r'<label[^>]+?for="(password)"', webpage, 'password field',
+                              fatal=False, default=None):
+            # Password protected
+            webpage, urlh = self._verify_video_password(webpage, urlh.geturl(), video_id)
```

Comment:
Why use `urlh.geturl()`? If just `url` is insufficient, add test for it
```suggestion
            webpage, urlh = self._verify_video_password(webpage, url, video_id)
```

Reply:
I can not remember, I think I had URLs that needed it (years ago when I created the extractor). But I will simplify it for now, and if we need it again I will rework it....